usr/share/rear/finalize/GNU/Linux/320_migrate_network_configuration_files.sh

```diff
@@ -102,9 +102,9 @@ function valid_restored_file_for_patching () {
 # FIXME: The following code fails if file names contain characters from IFS (e.g. blanks),
 # see https://github.com/rear/rear/pull/1514#discussion_r141031975
 # and for the general issue see https://github.com/rear/rear/issues/1372
-for restored_file in $TARGET_FS_ROOT/etc/sysconfig/*/ifcfg-* $TARGET_FS_ROOT/etc/network/inter[f]aces $TARGET_FS_ROOT/etc/network/interfaces.d/* ; do
+for restored_file in $TARGET_FS_ROOT/etc/sysconfig/*/ifcfg-* $TARGET_FS_ROOT/etc/network/inter[f]aces $TARGET_FS_ROOT/etc/network/interfaces.d/* $TARGET_FS_ROOT/etc/NetworkManager/system-connections/*.nmconnection ; do
     network_config_file="$( valid_restored_file_for_patching "$restored_file" )" || continue
-    network_config_files+=( $network_config_file )
+    network_config_files+=( "$network_config_file" )
```

Comment:
@pcahyna 
only out of curiosity:
Are there network config files with `$IFS` characters
or other problematic characters in their file names
or did you add quoting only to be on the safe side?

Reply:
Since a longer time I am waiting for basic system files
with `$IFS` characters or other problematic characters
in their file names.

Hooray :-( now I know about the first real-world example.

The current ReaR code will break at very many places
when basic system vaules contain `$IFS` characters
or other problematic characters.

Basically this means all usage of strings to store
more than a single value must be replaced with arrays.
Doable but a huge pile of legwork.

It also means that such values must be properly marked
as as single value in user output messages,
e.g. instead of
```
File not found: this that
```
where it is not clear if `this that` is one or two files
better
```
File not found: 'this that'
```
which looks "interesting" when the value contains `'`
```
# files=( "Mike's letter" 'Say "Hello"' "File not found" )

# for f in "${files[@]}" ; do echo "File not found: '$f'" ; done
File not found: 'Mike's letter'
File not found: 'Say "Hello"'
File not found: 'File not found'
```
Sigh!

Reply:
> Basically this means all usage of strings to store more than a single value must be replaced with arrays. Doable but a huge pile of legwork.

yes, we should gradually change the code this way. It will help with consistency anyway if the variable is meant to be user-settable, because we ahave a lot of array configuration variables already and it is simpler for the user to know that variables that store multiple values are always arrays than to remember when the variable is an array and when it is a single string of space-separated values.

> It also means that such values must be properly marked as as single value in user output messages, e.g. instead of
> 
> ```
> File not found: this that
> ```
> 
> where it is not clear if `this that` is one or two files better
> 
> ```
> File not found: 'this that'
> ```

We have been already doing that AFAICT, at least that has been my experience with existing log or terminal messages - they are usually quoted this way and I tried to respect it.

> 
> which looks "interesting" when the value contains `'`
> 
> ```
> # files=( "Mike's letter" 'Say "Hello"' "File not found" )
> 
> # for f in "${files[@]}" ; do echo "File not found: '$f'" ; done
> File not found: 'Mike's letter'
> File not found: 'Say "Hello"'
> File not found: 'File not found'
> ```
> 
> Sigh!

True. I don't expect this to occur often, but it is true that the "saxonic genitive" makes it more likely to happen.

Reply:
I experimented with several variable expansion flags, hoping that they would lead to quoting of such strange characters, but I have not been successful:
```
$ network_config_files=( "Pavel's WIFI" )
$ for network_config_file in "${network_config_files[@]}" ; do echo "Wrote new MAC addresses and network interfaces in '$network_config_file'"; done
Wrote new MAC addresses and network interfaces in 'Pavel's WIFI'
$ for network_config_file in "${network_config_files[@]}" ; do echo "Wrote new MAC addresses and network interfaces in '${network_config_file:Q}'"; done
Wrote new MAC addresses and network interfaces in 'Pavel's WIFI'
$ for network_config_file in "${network_config_files[@]}" ; do echo "Wrote new MAC addresses and network interfaces in '${network_config_file:K}'"; done
Wrote new MAC addresses and network interfaces in 'Pavel's WIFI'
```
I want the code being correct and safe for any possible input, but in the case of code used purely for display / informational purposes I am willing to relax this requirement: I don't want to spend too much time and make the code more complicated / less readable just to get a better formatting of log messages in corner cases.

Reply:
@pcahyna 
I didn't mean my
```
looks "interesting" when the value contains '
```
part that we should implement sophisticated code
to make such corner cases look nicer in ReaR.

Simple enclosing values with `'` in user messages like
```
LogPrint "... '$variable' ..."
```
is sufficient.

It is also in general useful to clearly show
when `$variable` is empty or blanks.

Forthermore simple `'` is predictable what the output
will look like from plain looking at the code.
Because at least I cannot find right now where
`${var:Q}` and `${var:K}` are documented
(searching for Q or K is not very helpful ;-)
So far up to now I only found `${var@Q}` and `${var@K}` in
https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

Reply:
@jsmeix oops! I meant the stuff with `@`, but I used `:`! Now it looks better:
```
$ network_config_files=( "Pavel's WIFI" )
$ for network_config_file in "${network_config_files[@]}" ; do echo "Wrote new MAC addresses and network interfaces in '${network_config_file@Q}'"; done
Wrote new MAC addresses and network interfaces in ''Pavel'\''s WIFI''
$ for network_config_file in "${network_config_files[@]}" ; do echo "Wrote new MAC addresses and network interfaces in '${network_config_file@K}'"; done
Wrote new MAC addresses and network interfaces in ''Pavel'\''s WIFI''
```

Reply:
One should drop the single quotes then, as apparently `@Q` produces an always-quoted version:
```
$ network_config_files=( "Pavel's WIFI" "Pavels WIFI"   "WIFI" )
$ for network_config_file in "${network_config_files[@]}" ; do echo "Wrote new MAC addresses and network interfaces in ${network_config_file@Q}"; done
Wrote new MAC addresses and network interfaces in 'Pavel'\''s WIFI'
Wrote new MAC addresses and network interfaces in 'Pavels WIFI'
Wrote new MAC addresses and network interfaces in 'WIFI'
```
I would say this is simple enough to be used instead of the single quotes, but of course it has the disadvantage of diminished readability of the code.
More seriously, even in RHEL 7 this does not work (bash is too old), so I think I will drop the idea.

Reply:
For user info messages I find
```
... in 'Pavel'\''s WIFI'
```
much harder to understand what the value actually is than
```
... in 'Pavel's WIFI'
```
where I think the human's brain "fuzzy logic"
does "the right magic" to deduce that
the correct meaning is `Pavel's WIFI`